File: JavaScript/프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.js
function solution(n, w, num) {
    const min_h = Math.floor(n/w)
    const max_h = min_h + 1;
    const count_min = max_h * w - n;
    var cur_h = Math.floor((num - 1) / w);
    var cur_w;
    var min_idx = [];
    
    if (cur_h % 2 == 0) cur_w = (num - 1) % w;
    else cur_w = w*(cur_h + 1) - num;
    
    for (let i=0; i<count_min; i++) {
        if (max_h % 2 == 0) min_idx.push(i);
        else min_idx.push(w-i-1);
    }
    
    if (min_idx.includes(cur_w)) return min_h - cur_h;
    return max_h - cur_h;
}

File: Python3/프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.py
def solution(n, w, num):
    min_h = n // w
    max_h = min_h + 1
    count_min = max_h * w - n

    cur_h = (num - 1) // w

    if cur_h % 2 == 0:
        cur_w = (num - 1) % w
    else:
        cur_w = w * (cur_h + 1) - num

    min_idx = []
    for i in range(count_min):
        if max_h % 2 == 0:
            min_idx.append(i)
        else:
            min_idx.append(w - i - 1)

    if cur_w in min_idx:
        return min_h - cur_h
    return max_h - cur_h